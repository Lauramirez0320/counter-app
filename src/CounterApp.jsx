import Proptypes from "proptypes";
import React, { useReducer } from "react";

const initialState = {
  counter: 0,
};

const reducer = (state, action) => {
  switch (action.type) {
    case "INCREASE":
      return { counter: state.counter + 1 };
    case "DECREASE":
      return { counter: state.counter - 1 };
    case "RESET":
      return { counter: 0 };
    default:
      throw new Error("There's no type of action");
  }
};

const CounterApp = () => {
  const [state, dispatch] = useReducer(reducer, initialState);
  return (
    <>
      <h1>CounterApp</h1>
      <h2>{state.counter}</h2>
      <button
        onClick={() => {
          dispatch({ type: "INCREASE" });
        }}
      >
        +1
      </button>
      <button
        onClick={() => {
          dispatch({ type: "DECREASE" });
        }}
      >
        -1
      </button>
      <button
        onClick={() => {
          dispatch({ type: "RESET" });
        }}
      >
        Reset
      </button>
    </>
  );
};

CounterApp.prototype = {
  value: Proptypes.number,
};

export default CounterApp;
