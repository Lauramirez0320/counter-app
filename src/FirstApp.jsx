
const newMessageData = [{
  message: "Hola Mundo",
  title: "Laura",
}];

const newMessage = (props) => {
 
  return JSON.stringify(props)
}

const FirstApp = () => {
  // const { message } = newMessage;
  return (
    <>
      <h1>{newMessage(newMessageData)}</h1>
    </>
  );
};

export default FirstApp;
