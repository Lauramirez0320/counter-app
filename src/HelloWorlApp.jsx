import { useState } from "react";
import FirstApp from "./FirstApp";

function App() {
  const [count, setCount] = useState(0);

  return (
    <div className="App">
      <FirstApp />
    </div>
  );
}

export default App;
