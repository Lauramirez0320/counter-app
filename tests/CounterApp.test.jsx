import { render, screen, fireEvent } from "@testing-library/react";
import { CounterApp } from "../src/CounterApp";

describe("CounterApp component", () => {
    Button = ({ onClick, children }) => (
    <button onClick={onClick}>{children}</button>
  );

  test("Should match snapshot", () => {
    const { container } = render(<CounterApp />);

    expect(container).toMatchSnapshot();
  });

  test("calls onClick prop when clicked +1", () => {
    const handleClick = jest.fn();
    render(<Button onClick={handleClick}>+1</Button>);
    fireEvent.click(screen.getByText("+1"));
    
    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  test('Should trigger handleClick in -1 button', ()=>{
    render(<CounterApp />);
    const decreaseButton = screen.getByText("-1");
    fireEvent.click(decreaseButton);
    fireEvent.click(decreaseButton);
    expect( screen.getByText('-2') ).toBeTruthy();
  });
});
