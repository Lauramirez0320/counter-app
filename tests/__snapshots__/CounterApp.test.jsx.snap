// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterApp component Should match snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2
    role="counter"
  >
    0
  </h2>
  <button
    data-testid="increase-button"
  >
    +1
  </button>
  <button>
    -1
  </button>
  <button>
    Reset
  </button>
</div>
`;
